code/game/objects/items/devices/traitordevices.dm

```diff
@@ -245,3 +245,97 @@
+	. = ..()
+	. += "<span class='warning'>Using this item in quick succession may cause fatigue to the user!</span>"
+	if(times_used >= max_uses)
+		. += "<span class='notice'>[src] is out of charge.</span>"
+	if(times_used < max_uses)
+		. += "<span class='notice'>[src] has [max_uses-times_used] charges left.</span>"
+
+/obj/item/batterer/attack_self(mob/living/carbon/user)
+	if(user)
+		if(times_used >= max_uses)
+			to_chat(user, "<span class='danger'>The mind batterer has been burnt out!</span>")
+			return
+		if(!do_after_once(user, 2 SECONDS, target = src, allow_moving = TRUE, attempt_cancel_message = "You think it's best to save this for later."))
+			return
+
+	for(var/mob/living/M in oview(7, get_turf(src)))
+		if(issilicon(M))
+			M.Weaken(10 SECONDS)
+		else
+			M.Confused(30 SECONDS)
+		M.adjustBrainLoss(10)
+		to_chat(M, "<span class='danger'>You feel a sudden, electric jolt travel through yourself,</span>")
+		switch(rand(1, 10))
+			if(1)
+				M.apply_status_effect(STATUS_EFFECT_CLINGTENTACLE_BATTERER)
+				to_chat(M, "<span class='warning'>and your legs lock up for a moment!</span>")
+			if(2)
+				M.apply_status_effect(STATUS_EFFECT_PACIFIED_BATTERER)
+				to_chat(M, "<span class='warning'>and you feel an innate love for life for a fleeting moment!</span>")
+			if(3)
+				if(prob(80))
+					M.Druggy(45 SECONDS)
+					to_chat(M, "<span class='warning'>and you feel like, totally chill dude!</span>")
+				else
+					M.AdjustHallucinate(45 SECONDS)
+					to_chat(M, "<span class='warning'>and this doesn't feel like a good trip!</span>")
+			if(4)
+				if(prob(80))
+					M.EyeBlurry(25 SECONDS)
+					to_chat(M, "<span class='warning'>and something in the back of your head stings like hell!</span>")
+				else
+					M.EyeBlind(15 SECONDS)
+					to_chat(M, "<span class='warning'>and you can't see a goddamn thing!</span>")
+			if(5)
+				M.adjustStaminaLoss(40)
+				to_chat(M, "<span class='warning'>and a wave of tiredness washes over you!</span>")
+			else
+				to_chat(M, "<span class='danger'>but as soon as it arrives, it fades.</span>")
+		add_attack_logs(user, M, "Mind battered with [src]")
+
+	playsound(get_turf(src), 'sound/misc/interference.ogg', 50, TRUE)
+	times_used++
+	to_chat(user, "<span class='notice'>You trigger [src]. It has [max_uses-times_used] charges left.</span>")
+	if(times_used >= max_uses)
+		icon_state = "battererburnt"
+
+/obj/item/batterer/throw_impact(atom/hit_atom)
+	..()
+	if(times_used >= max_uses)
+		return
+	visible_message("<span class='notice'>[src] suddenly triggers, sending a shower of sparks everywhere!</span>")
+	do_sparks(4, FALSE, get_turf(src))
+	attack_self()
+
+/obj/item/batterer/emp_act(severity)
+	if(times_used >= max_uses)
+		return
+	visible_message("<span class='notice'>[src] explodes into a light show of colors!</span>")
+	if(severity == EMP_HEAVY)
+		attack_self()
+		times_used = max_uses - 1
+		attack_self()
+	else
+		times_used = max_uses - 1
+		attack_self()
```

Comment:
Could you transfer all the code from attack_self() into another proc, then you can call that instead of using attack_self(). It makes it a lot easier to search through code

Reply:
sure